src/cargo/util/toml_mut/manifest.rs

```diff
@@ -537,9 +551,14 @@ fn non_existent_table_err(table: impl std::fmt::Display) -> anyhow::Error {
 
 fn non_existent_dependency_err(
     name: impl std::fmt::Display,
-    table: impl std::fmt::Display,
+    search_table: impl std::fmt::Display,
+    found_table: Option<impl std::fmt::Display>,
 ) -> anyhow::Error {
-    anyhow::format_err!("the dependency `{name}` could not be found in `{table}`.")
+    let mut msg = format!("the dependency `{name}` could not be found in `{search_table}`.");
+    if let Some(found_table) = found_table {
+        msg.push_str(&format!(" But it was found in `{found_table}`.",));
```

Comment:
nit: Starting with "But" feels a little abrupt.  Maybe
```
" It is present in `{found_table}`."
```

Maybe even remove the periods and change it to
```
"; it is present in `{found_table}`"
```

Reply:
Updated.